proselint/command_line.py

```diff
@@ -1,185 +1,183 @@
+        sys.exit(0)
 
-    if time:
-        # click.echo(timing_test())
-        print("This option does not work for the time being.")
-        return
+    if output_format in Output.names():
+        config["output_format"] = output_format
 
-    # In debug or clean mode, delete cache & *.pyc files before running.
-    if debug or clean:
-        clear_cache()
+    if benchmark:
+        run_benchmark()
+        sys.exit(0)
 
     # Use the demo file by default.
     if demo:
+        log.info("Demo-mode activated")
         paths = [demo_file]
 
-    # Expand the list of directories and files.
-    filepaths = extract_files(list(paths))
-
-    # Lint the files
-    num_errors = 0
-
-    # Use stdin if no paths were specified
-    if len(paths) == 0:
-        filepaths.append('-')
-
-    for fp in filepaths:
-        if fp == '-':
-            fp = '<stdin>'
-            f = sys.stdin
-        else:
-            try:
-                f = click.open_file(fp, 'r', "utf-8", "replace")
-            except Exception:
-                traceback.print_exc()
-                sys.exit(2)
-        errors = lint(f, debug, config)
-        num_errors += len(errors)
-        print_errors(fp, errors, output_json, compact)
-
-    # Return an exit code
-    close_cache_shelves()
-    if num_errors > 0:
-        sys.exit(1)
-    else:
-        sys.exit(0)
-
-
-def extract_files(files):
-    """Expand list of paths to include all text files matching the pattern."""
-    expanded_files = []
-    legal_extensions = [".md", ".txt", ".rtf", ".html", ".tex", ".markdown"]
-
-    for f in files:
-        # If it's a directory, recursively walk through it and find the files.
-        if os.path.isdir(f):
-            for dir_, _, filenames in os.walk(f):
-                for filename in filenames:
-                    fn, file_extension = os.path.splitext(filename)
-                    if file_extension in legal_extensions:
-                        joined_file = os.path.join(dir_, filename)
-                        expanded_files.append(joined_file)
-
-        # Otherwise add the file directly.
-        else:
-            expanded_files.append(f)
-
-    return expanded_files
-
-
-if __name__ == '__main__':
+    # prepare list
+    if paths is None:
+        paths = []
+    if isinstance(paths, str):
+        paths = [Path(paths)]
+    if isinstance(paths, Path):
+        paths = [paths]
+    if isinstance(paths, (list, tuple)):
+        paths = [Path(path) for path in paths]
+    log.debug("Paths to lint: %s", paths)
+
+    ts_start = time.time()
+    results = tools.lint_path(paths, config)
+    duration = time.time() - ts_start
+
+    error_sum = 0
+    for _file, _errors in results.items():
+        output_errors(_errors, config, _file)
+        error_sum += len(_errors)
+
+    log.info(
+        "Found %d lint-warnings in %.3f s (%d files, %.2f kiByte)",
+        error_sum,
+        duration,
+        len(results),
+        tools.last_char_count / 1024,
```

Comment:
```suggestion
    log.info(
        "Found {error_sum} lint warnings in {duration:.3f}s ({len(results)} files, {tools.last_char_count / 1024:.2f} kiB)",
```
It should be something like this.

Reply:
i also like faststrings - but it really shouldn't :)

Reply:
Is there no other way to lazily construct a string?